src/main/java/org/xstefank/model/WhitelistProcessing.java

```diff
@@ -0,0 +1,131 @@
+        this.config = config;
+    }
+
+    WhitelistProcessing(TemplateChecker templateChecker, FormatConfig config, String dirName) {
+        USER_LIST = new PersistentList(dirName, Utils.USERLIST_FILE_NAME);
+        ADMIN_LIST = new PersistentList(dirName, Utils.ADMINLIST_FILE_NAME);
+        LOGS = new PersistentList(dirName, Utils.GITHUB_CACHE_FILE_NAME);
+
+        this.issueCommands = config.getWhitelist().getIssueCommands();
+        this.pullRequestCommands = config.getWhitelist().getPullRequestCommands();
+        this.templateChecker = templateChecker;
+        this.config = config;
+    }
+
+    public void process(JsonNode payload) {
+        if (!payload.get(Utils.ACTION).asText().contains("deleted")) {
+            String authorUsername = payload.get(Utils.ISSUE).get(Utils.USER).get(Utils.LOGIN).asText();
+            String commUsername = payload.get(Utils.COMMENT).get(Utils.USER).get(Utils.LOGIN).asText();
+            String message = payload.get(Utils.COMMENT).get(Utils.BODY).asText();
+            String commentId = payload.get(Utils.COMMENT).get(Utils.ID).asText();
+            runWhitelistOperation(payload, authorUsername, commUsername, message, commentId);
+        } else {
+            deleteLog(payload);
+        }
+    }
+
+    public boolean isPrAuthorWhitelisted(JsonNode payload) {
+        String username = payload.get(Utils.PULL_REQUEST).get(Utils.USER).get(Utils.LOGIN).asText();
+        return USER_LIST.hasElement(username);
+    }
+
+    private void runWhitelistOperation(JsonNode payload, String authorUsername, String commUsername,
+                                       String message, String commentId) {
+        if (payload.get(Utils.ISSUE).has(Utils.PULL_REQUEST)) {
+            processWithPrCommands(payload, authorUsername, commUsername, message, commentId);
+        } else {
+            processWithIssueCommands(authorUsername, commUsername, message, commentId);
+        }
+    }
+
+    private void processWithPrCommands(JsonNode payload, String authorUsername, String commUsername, String message, String commentId) {
+        if (message.matches(pullRequestCommands.getAddUserCommand()) && !LOGS.hasElement(commentId)
+                && ADMIN_LIST.hasElement(commUsername)) {
+            USER_LIST.addElement(authorUsername);
+            LOGS.addElement(commentId);
+            sendStatus(payload);
+        }
+
+        if (message.matches(pullRequestCommands.getRemoveUserCommand()) && !LOGS.hasElement(commentId)
+                && ADMIN_LIST.hasElement(commUsername) && USER_LIST.hasElement(authorUsername)) {
+            USER_LIST.removeElement(authorUsername);
+            LOGS.addElement(commentId);
+        }
+
+        if (message.matches(pullRequestCommands.getRetestCommand()) && !LOGS.hasElement(commentId)
+                && ADMIN_LIST.hasElement(commUsername) && USER_LIST.hasElement(authorUsername)) {
+            LOGS.addElement(commentId);
+            sendStatus(payload);
+        }
+    }
```

Comment:
I don't like the approach you've taken here. The commands should be configurable, you've only hardcoded 3 possible commands so far. We would like to have an option to define for users some form of execution if the command is included